src/nimble_contact_manager.cc

```diff
@@ -351,88 +351,85 @@ namespace nimble {
     using face_iterator_t = std::vector< std::vector<int> >::iterator;
     using face_id_iterator_t = std::vector<int>::iterator;
 
-    int mpi_rank, num_ranks;
+    int mpi_rank = 0, num_ranks = 1;
     MPI_Comm_rank(MPI_COMM_WORLD, &mpi_rank);
     MPI_Comm_size(MPI_COMM_WORLD, &num_ranks);
 
-    std::cout << "Removing internal skin faces on rank " << mpi_rank << '\n';
-
-    int num_nodes_in_face = 4;
+    constexpr int num_nodes_in_face = 4;
 
-    const int* genesis_node_global_ids = mesh.GetNodeGlobalIds();
+    const int *genesis_node_global_ids = mesh.GetNodeGlobalIds();
     std::vector<int> face_global_ids;
-    face_global_ids.reserve(mesh.GetNumNodeGlobalIds());
-    for (auto& face : faces) {
-      for (int i : face) {
-        face_global_ids.push_back(genesis_node_global_ids[i]);
-      }
-    }
-
-    std::vector<std::vector<int>> global_id_map;
-    global_id_map.reserve( faces.size() );
-    for ( auto &&face : faces ) {
-      std::vector< int > fvec;
-      fvec.reserve( face.size() );
-      for ( auto &&node : face )
-        fvec.push_back(genesis_node_global_ids[node]);
-      global_id_map.emplace_back(std::move(fvec));
+    face_global_ids.reserve(num_nodes_in_face * faces.size());
+    std::map< std::vector<int>, int > faceList;
```

Comment:
Out of curiosity, is an unordered_map any faster? I know it has some performance issues with hash collisions but it may give even faster search time

Reply:
We have CI for MPI + ArborX up to 4 processors for the original version of `sphere_plate_contact`. And it passes the test. I mainly looked at the refined `sphere_plate_contact` with 2, 4, 8, and 16 processors. You are right that a unit test would be nice.

Reply:
As far as the unordered_map is concerned, I was concerned about the potential O(n) worst case and about creating a hash function for an object `std::vector<int>` of length 4. I was happy with the speed-up up to 16 processors. If your problem with ~ 500 ranks remains slow, maybe the unordered_map would be worth exploring.